File: api/src/model/UserConfig.js
const mongoose = require("mongoose");

const UserConfig = new mongoose.Schema({
	config: {
		type: Object,
		required: true,
		properties: {
			fontSize: { type: Number },
		},
	},
	user: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
});

module.exports = mongoose.model("UserConfig", UserConfig);

File: spring/user/src/main/java/com/schaefersm/chartissimo/model/UserConfig.java
package com.schaefersm.chartissimo.model;

import java.util.HashMap;

import javax.persistence.Id;

import org.bson.types.ObjectId;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document(collection = "userconfigs")
public class UserConfig {

    @Id
    private String _id;
    private ObjectId user;
    private Config config;

    public UserConfig(Config config, ObjectId user) {
        this.config = config;
        this.user = user;
    }

}